docs/spec/v1beta1/imagerepositories.md

```diff
@@ -110,6 +115,50 @@ kubectl create secret generic $SECRET_NAME \
   --from-file=caFile=ca.crt
 ```
 
+### Allow cross-namespace references
+
+To grant access to an `ImageRepository` for policies in other namespaces, the owner of the `ImageRepository`
+has to specify a list of label selectors that match the namespaces of the `ImagePolicy` objects.
+
+```yaml
+apiVersion: image.toolkit.fluxcd.io/v1beta1
+kind: ImageRepository
+metadata:
+  name: app1
+  namespace: apps
+spec:
+  interval: 5m
+  image: docker.io/org/image
+  secretRef:
+    name: regcred
+  accessFrom:
+  - namespaceSelector:
+      matchLabels:
+        kubernetes.io/metadata.name: flux-system
+```
+
+**Note** that the `kubernetes.io/metadata.name` is a readonly label added by Kubernetes >= 1.21
+automatically on namespaces. If you're using an older version of Kubernetes, please set labels
+on the namespaces where the `ImagePolicy` are.
+
+The above definition, allows for `ImagePolicy` in the `flux-system` namespace
```

Comment:
Is it a realistic scenario to have the `ImageRepository` in the namespace `apps`, but the policy objects in `flux-system`? Is that because the likelihood is people will have an automation object in flux-system so it can access the "system" `GitRepository`? (If so, let's remember to rewrite the examples once it's possible to do something different).

Reply:
Yes it boils down to the secret that `flux bootstrap github --read-write-key` creates that no one wants to copy it to other namespaces.